File: 144.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> preorderTraversal(TreeNode root) {
        Stack<TreeNode> st = new Stack<>();
        List<Integer> ans = new ArrayList<>();

        if (root==null){
            return ans;
        }

        st.push(root);

        while(!st.empty()){
            TreeNode tmp = st.pop();
            ans.add(tmp.val);

            // тут важно сначала добавить в стек правую, чтобы левая нода была первой

            if(tmp.right != null){
                st.push(tmp.right);
            }

            if(tmp.left != null){
                st.push(tmp.left);
            }


        }

        return ans;
    }
}

File: 144.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def preorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        stack = []
        ans = []

        if (root is None):
            return []

        stack.append(root)

        while (len(stack)>0):
            tmp = stack.pop()
            ans.append(tmp.val)

            if (tmp.right is not None):
                stack.append(tmp.right)

            if (tmp.left is not None):
                stack.append(tmp.left)



        return ans